_xtool/llclang/internal/parser/iparser.go

```
@@ -0,0 +1,36 @@
+package parser
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+)
+
+type Mode int
+
+func RunParseIntermediateFile(args []string) {
+	if len(args) != 2 {
+		fmt.Fprintln(os.Stderr, "RunParseIntermediateFile: lack of argument")
+		os.Exit(1)
+	}
+	filename := args[0]
+	var mode Mode
+	if args[1] != "" {
+		modeInt, err := strconv.Atoi(args[1])
+		if err != nil {
+			fmt.Fprintln(os.Stderr, "RunParseIntermediateFile: invalid mode")
+			os.Exit(1)
+		}
+		mode = Mode(modeInt)
+	}
+	err := parseIntermediateFile(filename, mode)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
```

It's better to throw an error instead. Then, handle the error in `main`.

---

Done in 5e12cc3956ee796c7f8aa5976140b6aa696bbb6d